common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -286,13 +302,29 @@ namespace com.keyman.keyboards {
 
     }
 
+    static sanitize(rawRow: LayoutRow) {
+      for(const key of rawRow.key) {
+        // Test for a trailing comma included in spec, added as null object by IE
+        // It has only ever appeared at the end of a row's spec.
+        if(key == null) {
+          rawRow.key.length = rawRow.key.length-1;
```

Comment:
TIL that `length` is a settable property and this does not excite me due to the [weirdness of what happens](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#array_methods_and_empty_slots) when you lengthen an array this way!

I did not know that "empty" was different to `undefined` in an array and now I cry.

Reply:
Me too - I was surprised when I saw it in this code block's original spot.  I could rework that to something a bit more intuitive if desired.

Reply:
It's fine as is. It works and is nicer than `splice` or other similar things...